test/behaviors/integrationVault.ts

```diff
@@ -0,0 +1,799 @@
+            );
+        });
+
+        describe("edge cases", () => {
+            it("reverts when vault's nft is 0", async () => {
+                await ethers.provider.send("hardhat_setStorageAt", [
+                    this.subject.address,
+                    "0x4", // address of _nft
+                    "0x0000000000000000000000000000000000000000000000000000000000000000",
+                ]);
+                await expect(
+                    this.subject.push(
+                        [this.usdc.address],
+                        [BigNumber.from(1)],
+                        []
+                    )
+                ).to.be.revertedWith(Exceptions.INIT);
+            });
+            it("reverts when owner's nft is 0", async () => {
+                let address = `0x${this.erc20RootVault.address
+                    .substr(2)
+                    .padStart(64, "0")}`;
+                await ethers.provider.send("hardhat_setStorageAt", [
+                    this.vaultRegistry.address,
+                    ethers.utils.keccak256(
+                        encodeToBytes(
+                            ["bytes32", "uint256"],
+                            [address, BigNumber.from(10)]
+                        )
+                    ), // setting _nftIndex[this.erc20RootVault.address] = 0
+                    "0x0000000000000000000000000000000000000000000000000000000000000000",
+                ]);
+                await expect(
+                    this.subject.push(
+                        [this.usdc.address],
+                        [BigNumber.from(1)],
+                        []
+                    )
+                ).to.be.revertedWith(Exceptions.NOT_FOUND);
+            });
+            it("reverts when tokens and tokenAmounts lengths do not match", async () => {
+                await expect(
+                    this.subject.push(
+                        [this.usdc.address],
+                        [BigNumber.from(1), BigNumber.from(1)],
+                        []
+                    )
+                ).to.be.revertedWith(Exceptions.INVALID_VALUE);
+            });
+            it("reverts when tokens are not sorted", async () => {
+                await expect(
+                    this.subject.push(
+                        [this.weth.address, this.usdc.address],
+                        [BigNumber.from(1), BigNumber.from(1)],
+                        []
+                    )
+                ).to.be.revertedWith(Exceptions.INVARIANT);
+            });
+            it("reverts when tokens are not unique", async () => {
+                await expect(
+                    this.subject.push(
+                        [this.usdc.address, this.usdc.address],
+                        [BigNumber.from(1), BigNumber.from(1)],
+                        []
+                    )
+                ).to.be.revertedWith(Exceptions.INVARIANT);
+            });
+            it("reverts when tokens not sorted nor unique", async () => {
```

Comment:
when tokens are not unique

Reply:
"when tokens are not unique" test goes above of that